CBE0055 Diabetes: Eye Exam

utsavpaudel opened the pull request:
## Describe your changes
CBE0055 Diabetes: Eye Exam Quality measure has been implemented as per the CMS document provided.


## How has this been tested?
Measure output has been manually tested with the CMS document's logic. Denominator, Numerator, and Exclusion model's result has been validated as to where it's derived from.
Refer to the sheet for details.
https://docs.google.com/spreadsheets/d/1csJNaTf0lpEJxgQML8zVKHcDDlVoltynDPBAWZuH8Co/edit?usp=sharing


## Reviewer focus
Please emphasize on the logic of `retinopathy_last_year` ,  `no_retinopathy_last_year` in the `numerator` model.
and `diabetics_codes` in the `denominator` model. 


## Checklist before requesting a review
- [ ] I have updated the version number in dbt_project.yml file to reflect the release number of this PR
- [x] I have updated the docs files (by running dbt docs generate/serve and copying the necessary files into the docs folder)
- [x] I have commented my code as necessary
- [ ] I have added at least one Github label to this PR
- [x] My code follows style guidelines
- [ ] (Optional) I have recorded a Loom to explain this PR


## (Optional) Gif of how this PR makes you feel


## Loom link

Resolves #195

utsavpaudel commented on models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055_numerator.sql:
```diff
@@ -0,0 +1,209 @@
+
+)
+
+, qualifying_patients_for_procedure as (
+
+    select 
+          procedures.patient_id
+        , procedure_date
+        , procedures.code_type
+        , procedures.code
+    from procedures
+    inner join {{ ref('quality_measures__int_cbe0055__performance_period') }} pp
+    on procedure_date between
+        performance_period_begin and performance_period_end
+
+    inner join retina_test_code
+        on procedures.code = retina_test_code.code
+            and procedures.code_type = retina_test_code.code_system
+
+)
```
```
, qualifying_patients_for_procedure as (

    select 
          procedures.patient_id
        , procedure_date
        , procedures.code_type
        , procedures.code
    from procedures
    inner join {{ ref('quality_measures__int_cbe0055__performance_period') }} pp
    on procedure_date between
        performance_period_begin and performance_period_end
    inner join retina_test_code
        on procedures.code = retina_test_code.code
            and procedures.code_type = retina_test_code.code_system
             and lower(retina_test_code.concept_name) in ( 
                        'diabetic retinal eye exam met' 
                      , 'retinal or dilated eye exam'
                      )

)

, qualifying_patients_with_ophthalm_consult as (

    select 
          procedures.patient_id
        , qualifying_patients_for_procedure.procedure_date
        , qualifying_patients_for_procedure.code_type
        , qualifying_patients_for_procedure.code
    from procedures
    inner join {{ ref('quality_measures__int_cbe0055__performance_period') }} pp
    on procedure_date between
        performance_period_begin and performance_period_end
    inner join retina_test_code
        on procedures.code = retina_test_code.code
            and procedures.code_type = retina_test_code.code_system
              and lower(retina_test_code.concept_name) = 'ophthalmological services'
    inner join qualifying_patients_for_procedure
      on qualifying_patients_for_procedure.patient_id = procedures.patient_id

)
```

utsavpaudel commented:
Hi @tuvaforrest/ @sarah-tuva, we revisited the measure's document again, and felt the mentioned changes should be incorporated to better match the description in the document. The numerator model's final patient count do not change with the change though. 
Please let us know what you think. 
Thank you.

sarah-tuva commented:
Resolves #195

sarah-tuva commented on models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055_numerator.sql:
```diff
@@ -0,0 +1,238 @@
+    From {{ref('quality_measures__value_sets')}}
+    where lower(concept_name) in  (
+          'diabetic retinopathy'
+        , 'ophthalmological services'
```
This concept should be moved to the visit code logic in the denominator. You can confirm the codes in this concept are part of the "Patient encounter during the performance period" logic for the denominator in the measure definition.

sarah-tuva commented on models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055_numerator.sql:
```diff
@@ -0,0 +1,238 @@
+
+)
+
+, qualifying_patients_with_ophthalm_consult as (
+
+    select 
+        qualifying_procedure.patient_id,
+        qualifying_procedure.procedure_date,
+        qualifying_procedure.code_type,
+        qualifying_procedure.code
+    from qualifying_procedure
+    inner join qualifying_patients_for_procedure
+        on qualifying_procedure.patient_id = qualifying_patients_for_procedure.patient_id
+        and lower(qualifying_procedure.concept_name) = 'ophthalmological services'
```
Not needed once the concept "ophthalmological services" is moved to the denominator.

sarah-tuva commented on models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055_numerator.sql:
```diff
@@ -0,0 +1,238 @@
+
+)
+
+, retinopathy_conditions as (
```
The with/without retinopathy numerator criteria in this measure is confusing. If you read the numerator description, it is saying the same thing. A diabetic patient must have an eye exam during the measurement period. 

I would actually remove all of the retinopathy logic and just qualify any patient from the denominator that had an eye exam during the measurement period as meeting the numerator criteria.

sarah-tuva commented on models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055__denominator.sql:
```diff
@@ -0,0 +1,208 @@
+        , code_system
+    from {{ ref('quality_measures__value_sets') }}
+    where lower(concept_name) in (
+          'diabetes visit'
```
This concept should be included with the other visit codes CTE since these are procedure codes. You can confirm that the codes in this concept are part of the "Patient encounter during the performance period" logic for the denominator in the measure definition.

sarah-tuva commented on models/quality_measures/intermediate/cbe0055_diabetes_eye_exam/quality_measures__int_cbe0055__denominator.sql:
```diff
@@ -0,0 +1,208 @@
+        , 'preventive care services initial office visit, 18 and up'
+        , 'annual wellness visit'
+        , 'telephone visits'
+        , 'emergency department evaluation and management visit'
```
Remove this concept from the denominator visit_codes logic.

utsavpaudel replied:
Can you please share the reason as to why they're removed? This is in the `value set concepts` seeds for this measure.

sarah-tuva replied:
@utsavpaudel sorry for the confusion! Only 'emergency department evaluation and management visit' should be removed from the list. This concept should not be used in the denominator. It's only used for the acute inpatient exclusion logic.

sarah-tuva reviewed (changes requested):
Hi @utsavpaudel, I left a few requests for changes. Thank you so much for your work on this!

utsavpaudel commented:
Hello @sarah-tuva, requested changes have been incorporated. Thank you for the suggestions.

utsavpaudel commented:
Hi @sarah-tuva, latest enhancement from https://github.com/tuva-health/tuva/pull/461 has uploaded all existing seed files into s3 and removed content from dbt.
Resolving merge conflicts, we also removed all the seeds for quality measures.
Looks like seeds for this measure has not yet been uploaded to S3, so please upload the seeds as necessary before merging this PR.
Thank you.

sarah-tuva commented:
> Hi @sarah-tuva, latest enhancement from #461 has uploaded all existing seed files into s3 and removed content from dbt. Resolving merge conflicts, we also removed all the seeds for quality measures. Looks like seeds for this measure has not yet been uploaded to S3, so please upload the seeds as necessary before merging this PR. Thank you.

Good call out. Thanks @utsavpaudel! I'll work on this today and finalize the PR review/testing.

sarah-tuva commented:
@utsavpaudel I attempted to commit to your PR, but I don't think it worked. You'll need to update the [dbt_project.yml](https://github.com/tuva-health/tuva/blob/main/dbt_project.yml#L186) file in the repo root with the S3 path of the upcoming version. I uploaded the latest version of the seeds to this path in S3: `tuva-public-resources/versioned_value_sets/0.8.2`

Here's what that will look like in your branch:
![image](https://github.com/tuva-health/tuva/assets/106100509/9edd1f3b-8a07-4c8d-8808-16fd14cafefc)

You can leave the other seeds as they are. We will create a separate PR updating the others when we are ready to create a new release. Let me know if you have questions about this. And thanks for bearing with us while we figure out this new workflow with versioned seeds.

utsavpaudel commented:
Hey @sarah-tuva, mentioned path has been updated. Please have a look again.
